arch/z80/src/common/z80_assert.c

```diff
@@ -33,91 +33,11 @@
 #include <nuttx/arch.h>
 #include <nuttx/board.h>
 #include <nuttx/syslog/syslog.h>
```

Comment:
remove the unneeded header files:
nuttx/syslog/syslog.h
nuttx/irq.h
stdint.h
stdlib.h
assert.h